website/docs/docs/building-a-dbt-project/building-models/python-models.md

```diff
@@ -0,0 +1,378 @@
+:::
+
+### Data frame API + syntax
+
+Over the past decade, most people writing data transformations in Python have adopted <Term id="dataframe">data frame</Term> as their common abstraction. dbt follows this convention by returning `ref()` and `source()` as data frames, and it expects all Python models to return a data frame.
+
+A data frame is a two dimensional data structure (rows and columns). It supports convenient methods for transforming that data, creating new columns from calculations performed on existing columns. It also offers convenient methods for previewing data while developing locally or in a notebook.
+
+That's about where the agreement ends. There are numerous frameworks with their own syntaxes and APIs for data frames. The [Pandas](https://pandas.pydata.org/docs/) library offered one of the original data frame APIs, and its syntax is the most common to learn for new data professionals. Most newer data frame APIs offer some degree of compatibility with Pandas-style syntax, though few can offer perfect interoperability. This is true for both Snowpark and PySpark, which have their own DataFrame APIs.
+
+When developing a Python model, you will find yourself asking these questions:
+
+**Why Pandas?** The most common API for data frames. Makes it easy to explore sampled data and develop transformations locally. For small datasets, you can "promote" your code as-is into dbt models, and run in production.
+
+**Why _not_ Pandas?** Performance. Pandas runs "single-node" transformations, which cannot benefit from the parallelism and distributed compute offered by modern data warehouses. This quickly becomes a problem as you operate on larger datasets. Some data platforms do support optimizations for code written using Pandas' data frame API, preventing the need for major refactors. For example, ["Pandas on PySpark"](https://spark.apache.org/docs/latest/api/python/getting_started/quickstart_ps.html) offers support for 95% of Pandas functionality using the same API.
+
+:::info ❔ Our questions
+When developing a new dbt Python model, should we recommend Pandas-style syntax for rapid iteration, and then refactor? Which open source libraries provide compelling abstractions across different data engines and vendor-specific APIs? Should dbt attempt to play a longer-term role in standardizing across them?
+:::
+
+### Limitations
+
+Python models have capabilities that SQL models do not. They also have some drawbacks compared to SQL models:
+- **Time and cost.** Python models are slower to run than SQL models, and the cloud resources that run them can be more expensive. Running Python requires more general-purpose compute, and in some cases that compute may live on a separate service or architecture from your SQL models. However, we believe that deploying Python models via dbt is **dramatically** faster and cheaper than spinning up separate tooling and infrastructure to orchestrate Python transformations in production.
```

Comment:
In general I think the cost for the actual execution of the model is going to be similar, but we will save the cost to have separate orchestration and have better integration with the rest of your data that created by dbt project

Reply:
Yup - going to qualify that I mean cost from the perspective of the human maintaining this stuff